src/marqo/tensor_search/tensor_search.py

```diff
@@ -360,8 +360,14 @@ def add_documents(config: Config, add_docs_params: AddDocsParams):
     except errors.IndexNotFoundError:
         raise errors.IndexNotFoundError(f"Cannot add documents to non-existent index {add_docs_params.index_name}")
 
-    if len(add_docs_params.docs) == 0:
+    doc_count = len(add_docs_params.docs)
+
+    if doc_count == 0:
         raise errors.BadRequestError(message="Received empty add documents request")
+    elif doc_count > utils.read_env_vars_and_defaults_ints(EnvVars.MARQO_MAX_ADD_DOCS_COUNT):
+        raise errors.BadRequestError(message=f"Number of docs in add documents request ({doc_count}) exceeds limit of {utils.read_env_vars_and_defaults(EnvVars.MARQO_MAX_ADD_DOCS_COUNT)}. "
+                                     f"This limit can be increased by setting the environment variable `{EnvVars.MARQO_MAX_ADD_DOCS_COUNT}`."
+                                     f"If using the Python client, use `client_batch_size`. See https://marqo.pages.dev/1.3.0/API-Reference/documents/#body for more details.")
```

Comment:
Good idea mentioning `client_batch_size`. However, can we make this sentence more helpful? 
```
If using the Python client, use `client_batch_size`. 
```
For example:
```
If using the Python client, set the add_document method's `client_batch_size` parameter to break up a large sequence of documents into smaller batches.
```

Reply:
done